### PR TITLE
QSignature Compare tidy-up

### LINKED ISSUE #122: qsignature compare tidy-up
`Qsignature` has a number of different compare classes, and it would be great if they could be merged into one.
Here is the list of current Compare classes:

- `Compare` (awaiting approval in PR #121 )
- `CompareGenotypeBespoke` (deprecated in PR #121 )
- `SignatureCompareRelatedSimpleGenotype` (deprecated in PR #121 )
- `CompareIlluminaData`
- `CompareRGGenotype`
- `QSigCompare`
- `QSigCompareDistance`
- `SignatureCompareRelated`
- `SignatureCompareRelatedSimple`
- `SignatureCompareRelatedSimpleGenotypeMT`

Thats a lot of comparison classes!

It would be great to find out which ones are no longer used and deprecate them.
It would also be great to see if the remaining comparisons can be merged into a single compare class.

**Describe the solution you'd like**
Ideally all comparisons that can be merged into the single Compare class should be merged. There may be some specific comparisons that won't be able to be merged, and in those cases, they need to be appropriately named and documented.

**Describe alternatives you've considered**
The alternative is to leave the 10 comparison classes alone, and to thoroughly document each one so that the end users know which one to pick.

### PR DESCRIPTION
**Pull request by holmeso**
# Description

There are currently many compare classes within the `qsignature` project that do very similar things.
This PR has deprecated many of them, with a view to deleting them in a future version.
One class has been renamed. `CompareRGGenotype` is now `CompareRG` (the genotype part was meaning less).
Documentation has also been added to help the user understand what the class does.
And finally, changes to adhere to the checkstyle plugin used have been incorporated.

Fixes #122 

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

The project has been built with no issues

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: qsignature/src/org/qcmg/sig/QSigCompare.java [hunk 1]
```diff
@@ -388,10 +418,18 @@ public static double getDiscretisedValue(double initialValue) {
 		String snpFile = null;
 		for (Iterator<String> iter = header.iterator() ; iter.hasNext() ; ) {
 			String headerLine = iter.next();
-			if (headerLine.contains("patient_id")) patient = headerLine.substring(headerLine.indexOf("=") + 1);  
-			if (headerLine.contains("library")) library = headerLine.substring(headerLine.indexOf("=") + 1);  
-			if (headerLine.contains("input_type")) inputType = headerLine.substring(headerLine.indexOf("=") + 1);
-			if (headerLine.contains("snp_file")) snpFile = headerLine.substring(headerLine.indexOf("=") + 1);
+			if (headerLine.contains("patient_id")) {
+				patient = headerLine.substring(headerLine.indexOf("=") + 1);
```
Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/QSigCompare.java]: remove these empty lines

File: qsignature/src/org/qcmg/sig/QSigCompare.java [hunk 2]
```diff
@@ -388,10 +418,18 @@ public static double getDiscretisedValue(double initialValue) {
 		String snpFile = null;
 		for (Iterator<String> iter = header.iterator() ; iter.hasNext() ; ) {
 			String headerLine = iter.next();
-			if (headerLine.contains("patient_id")) patient = headerLine.substring(headerLine.indexOf("=") + 1);  
-			if (headerLine.contains("library")) library = headerLine.substring(headerLine.indexOf("=") + 1);  
-			if (headerLine.contains("input_type")) inputType = headerLine.substring(headerLine.indexOf("=") + 1);
-			if (headerLine.contains("snp_file")) snpFile = headerLine.substring(headerLine.indexOf("=") + 1);
+			if (headerLine.contains("patient_id")) {
+				patient = headerLine.substring(headerLine.indexOf("=") + 1);  
+			}
```
Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/QSigCompare.java]: remove empty lines

File: qsignature/src/org/qcmg/sig/QSigCompare.java [hunk 3]
```diff
@@ -388,10 +418,18 @@ public static double getDiscretisedValue(double initialValue) {
...
+			if (headerLine.contains("library")) {
```
Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/QSigCompare.java]: remove empty line here

File: qsignature/src/org/qcmg/sig/QSigCompareDistance.java [hunk 4]
```diff
@@ -45,6 +45,11 @@
 
 /**
  * The Class QSigCompareDistance.
+ * 
+ * This uses the Euclidean distance method to compare positions.
+ * This class has been superseded by the Compare class and so is deprecated. 
+ * 
+ * @deprecated
```
Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/QSigCompareDistance.java]: please move  "deprecated" annotation begin of the class, as this class should be deprecated or delete

File: qsignature/src/org/qcmg/sig/SignatureCompareRelated.java [hunk 5]
```diff
@@ -248,12 +248,15 @@ private int engage() throws Exception {
 			logger.info("No suspicious results found");
 		} else {
 			logger.info("SUMMARY:");
-			for (String s : suspiciousResults) logger.info(s);
+			for (String s : suspiciousResults) {
+				logger.info(s);
+			}
 			//email();
```
Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/SignatureCompareRelated.java]: remove this comment

File: qsignature/src/org/qcmg/sig/SignatureCompareRelatedSimple.java [hunk 6]
```diff
@@ -38,6 +37,7 @@
  *  
  * @author o.holmes
  *
+ *@deprecated Superseded by Compare
```
Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/SignatureCompareRelatedSimple.java]: again, this class should be deprecated now and delete it on next version.

File: qsignature/src/org/qcmg/sig/SignatureCompareRelatedSimpleGenotypeMT.java [hunk 7]
```diff
@@ -37,13 +34,18 @@
...
+
 /**
  * This class gets a list of all .qsig.vcf files from the supplied path.
  * It then performs a comparison between them all, regardless of whether they are bam or snp chip files
  * An xml output file is produced
  * If any comparison scores are less than the cutoff, they are added to a list, which is then emailed to interested parties informing them of the potential problem files
  *  
  * @author o.holmes
+ * 
+ * @Deprecated Superseded by Compare
```
Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/SignatureCompareRelatedSimpleGenotypeMT.java]: again, deprecated now if it is planed to delete on next version

File: qsignature/src/org/qcmg/sig/SignatureFindRelated.java [hunk 8]
```diff
@@ -41,6 +41,8 @@
  * 
  * 
  * @author o.holmes
+ * 
+ * @deprecated Superseded by Compare
```
Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/SignatureFindRelated.java]: again, please deprecated now.

File: qsignature/src/org/qcmg/sig/model/Comparison.java [hunk 9]
```diff
@@ -107,15 +84,18 @@ public String toString() {
...
-//		sb.append(", ").append(numberOfCalculations);
 		return sb.toString();
 	}
 
 	@Override
 	public int compareTo(Comparison o) {
 		
-		if (score < o.score) return -1;
-		if (score > o.score) return 1;
+		if (score < o.score) {
```
Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/model/Comparison.java]: suggest to change:
if (score == o.score) {
    return overlapCoverage - o.overlapCoverage;
}
return (score> o.score )? 1 : -1;

File: qsignature/src/org/qcmg/sig/util/ComparisonUtil.java [hunk 10]
```diff
@@ -6,9 +6,6 @@
  */
 package org.qcmg.sig.util;
```
Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/util/ComparisonUtil.java]: overlapCoverage - o.overlapCoverage;

Comment (ChristinaXu2017) [inline qsignature/src/org/qcmg/sig/util/ComparisonUtil.java]: have you deleted the old file and write a new one? they look very different in the comparison window.  But I think they do the same work for both old and new files. Is the related unit test updated?

Comment (holmeso) [reply qsignature/src/org/qcmg/sig/util/ComparisonUtil.java]: Not sure what happened - this file hasn't really changed...

File: qsignature/src/org/qcmg/sig/model/Comparison.java (hunk 9 above)
Comment (holmeso) [reply qsignature/src/org/qcmg/sig/model/Comparison.java]: done

File: qsignature/src/org/qcmg/sig/SignatureFindRelated.java (hunk 8 above)
Comment (holmeso) [reply qsignature/src/org/qcmg/sig/SignatureFindRelated.java]: done

File: qsignature/src/org/qcmg/sig/SignatureCompareRelatedSimpleGenotypeMT.java (hunk 7 above)
Comment (holmeso) [reply qsignature/src/org/qcmg/sig/SignatureCompareRelatedSimpleGenotypeMT.java]: done

File: qsignature/src/org/qcmg/sig/SignatureCompareRelatedSimple.java (hunk 6 above)
Comment (holmeso) [reply qsignature/src/org/qcmg/sig/SignatureCompareRelatedSimple.java]: done

File: qsignature/src/org/qcmg/sig/SignatureCompareRelated.java (hunk 5 above)
Comment (holmeso) [reply qsignature/src/org/qcmg/sig/SignatureCompareRelated.java]: done

File: qsignature/src/org/qcmg/sig/QSigCompareDistance.java (hunk 4 above)
Comment (holmeso) [reply qsignature/src/org/qcmg/sig/QSigCompareDistance.java]: done

**Comment by holmeso** [comment]
Thanks for the comments - code has been modified as suggested

**Pull request merged**